### PR TITLE
feat(rootkeys): add new constructor for root key store

### PR DESCRIPTION
**Pull request by jack-w-shaw**
Previously, the only constructor we had for bakery.RootkeyStore took in a Backing, but if that backing also implements ContextBacking, we prefer that backing.

Changes being made in Juju mean tha we naturally wish to implement ContextBacking. Our db methods require a context. However, to make use of the existing NewStore constructor, we would need to implement stub methods without contexts.

We do not wish to expose these.

Add a new constructor which directly takes in a ContextBacking as it's backing. This means we will no longer need to implement stub context-less methods.

## QA Steps

In tests, all we wish to assert is that we can construct a functioning RootKeyStore without implementing `GetKey`, etc.

Add a unit test and a fake backing to test this.

```
go test ./bakery/dbrootkeystore/...
```

### CODE REVIEW

**Review by SimonRichardson — APPROVED** [review]
In the future, we might want to start dropping the Backing interface in favor of the ContextBacking.

**Review by kian99 — APPROVED** [review]

File: bakery/dbrootkeystore/rootkey_test.go [hunk 1]
```diff
@@ -422,30 +422,66 @@ func TestContextBackingTakesPrecedence(t *testing.T) {
 	c.Assert(key1, qt.DeepEquals, key2)
 }
 
-type contextBacking struct {
+type fullContextBacking struct {
 	b dbrootkeystore.Backing
 }
 
-func (b contextBacking) GetKey(id []byte) (dbrootkeystore.RootKey, error) {
+func (b fullContextBacking) GetKey(id []byte) (dbrootkeystore.RootKey, error) {
```
Comment (alesstimec) [inline bakery/dbrootkeystore/rootkey_test.go:429]: perhaps a godoc explaining why this is returning an error for posterity..

**Review by alesstimec — APPROVED** [review]
LGTM, but naming is becoming a bit of a mess..

**Comment by alesstimec** [comment]
> In the future, we might want to start dropping the Backing interface in favor of the ContextBacking.

Hopefully juju won’t be using macaroons much longer :)

**Pull request merged**